rspace/src/main/scala/coop/rchain/rspace/ReplayRSpace.scala

```diff
@@ -107,7 +97,11 @@ class ReplayRSpace[F[_]: Concurrent: ContextShift: Log: Metrics: Span, C, P, A,
       channelToIndexedDataList <- channels.traverse { c: C =>
                                    store
                                      .getData(c)
-                                     .map(_.iterator.zipWithIndex.filter(matches(comm)).toSeq)
+                                     .flatMap(
+                                       _.iterator.zipWithIndex
+                                         .to[immutable.Seq]
```

Comment:
Instead of converting to a `List`, `immutable.Seq` has instance for Traversable.